src/db_reference.rs

```diff
@@ -0,0 +1,39 @@
+use rocksdb::DB;
+use std::cell::RefCell;
+use std::sync::Arc;
+
+/// The type of a reference to a [rocksdb::DB] that is passed around the library.
+pub(crate) type DbReference = Arc<RefCell<DB>>;
+
+/// A wrapper around [DbReference] that cancels all background work when dropped.
+///
+/// All users of [rocksdb::DB] should use this wrapper instead to avoid keeping background threads
+/// alive after the database is dropped.
+#[derive(Clone)]
+pub(crate) struct DbReferenceHolder {
+    inner: Option<DbReference>,
+}
+
+impl DbReferenceHolder {
+    pub fn new(db: DB) -> Self {
+        Self {
+            inner: Some(Arc::new(RefCell::new(db))),
+        }
+    }
+
+    pub fn get(&self) -> Option<&DbReference> {
+        self.inner.as_ref()
+    }
+
+    pub fn close(&mut self) {
+        if let Some(db) = self.inner.take().and_then(Arc::into_inner) {
+            db.borrow_mut().cancel_all_background_work(true);
+        }
+    }
```

Comment:
Will cancel_all_background_work be called multiple times when we have multiple DBReferencreHolder? I thought we should call it only on dropping the last ref.

Reply:
@Congyuwang `cancel_all_background_work` gets run exactly once. `Arc::into_inner` guarantees that only the last reference will return `Some`.

`DBCell` is another option that could work here if you don't like the wrapping struct here.